ai_services/anomaly_detection/sensitivity_auto_tuner/standalone_univariate/README.md

```diff
@@ -0,0 +1,92 @@
+## Sensitivity Auto Tuning based on Labeled Data
+
+Anomaly Detection Univariate Inference Workflow API provides a sensitivity parameter to move the decision boundary based on which we classify whether a timestamp is an anomaly. In other words, it is a parameter to tradeoff between False Positive Rate and True Positive Rate.
+Other Anomaly Detection Service features also offer the sensitivity parameter and this solution can be extended to those features as well.
+
+
+### Problem Statement:
+
+Users of AD service can use the Sensitivity parameter to tune the model performance based on their desired False Positive Rate(FPR) and True Positive Rate(TPR). The default value for the sensitivity parameter is set to 0.5 which will give the user a balanced model. Increasing the sensitivity will make the model more sensitive and predict more anomalies. The Sensitivity parameter gives this control to the user on how they want the model to perform. In some use cases, a higher TPR would be absolutely necessary even if it comes at the cost of a higher FPR. Similarly, some use cases might demand the model to report as fewer False Positives as possible and would be tolerant towards missed alarms.
+
+The complication with this approach is that the user would need to make several calls while varying the sensitivity manually. Users cannot realistically try a large number of values and may not be able to converge to the best possible model performance. To simplify this problem for the users, we have developed a tool that automates this process and calculates the optimal Sensitivity value for a given dataset with respect to the desired success metrics. The users do not need to use this tool if the default model performance is already within their desired success metrics.
+
+Users can use this tool on a labeled dataset and obtain the optimal sensitivity value for that dataset. Then, while making inference calls in production, where the data pattern doesn't vary too much, they can provide this sensitivity value in the inference call request and expect the desired performance.
+
+### Requirements:
+
+The success metrics (TPR and FPR) are calculated by comparing predicted labels against actual labels. Hence, this tool has the following dataset format requirements:
+
+*   **timestamp**: ISO-160 format with monotonically increasing and de-duplicated timestamp
+*   **value**: Numerical column with
+*   **anomaly**: Values labeled as 0(non-anomalous) / 1(anomalous)
+
+Users can use the open-source [TagAnomaly](https://github.com/microsoft/TagAnomaly) tool to annotate the dataset. We have an instance of this tool hosted at [http://100.73.92.203:3838/](http://100.73.92.203:3838/). Following is an example of a compatible dataset. Users can also edit the script in case they need to use different column names for their specific use case.  
```

Comment:
Notify the users that even with using TagAnomaly tool, they need to populate their own binary 'anomaly' column before calling this script.

Reply:
Resolved